src/DependencyInjection/SonataUserExtension.php

```diff
@@ -369,4 +371,19 @@ private function registerSonataDoctrineMapping(array $config): void
                 ]])
         );
     }
+
+    /**
+     * We MUST remove this method when support for "friendsofsymfony/user-bundle" is dropped
+     * or adapted to work with "doctrine/common:^3".
+     */
+    private function createDoctrineCommonBackwardCompatibilityAliases(): void
+    {
+        if (!interface_exists(\Doctrine\Common\Persistence\ObjectManager::class)) {
+            class_alias(\Doctrine\Persistence\ObjectManager::class, \Doctrine\Common\Persistence\ObjectManager::class);
+        }
+
+        if (!class_exists(\Doctrine\Common\Persistence\Event\LifecycleEventArgs::class)) {
+            class_alias(\Doctrine\Persistence\Event\LifecycleEventArgs::class, \Doctrine\Common\Persistence\Event\LifecycleEventArgs::class);
+        }
+    }
```

Comment:
:thinking: why is this needed? `doctrine/persistence` already comes with a persistence layer that does the opposite…

Reply:
FosUser is using some `\Doctrine\Common\Persistence` classes, so if you're using persistence 2.0, you don't have them, isn't it ?

Reply:
Indeed, but this is an issue (or rather several issues) that should be fixed there:
- FosUser should declare its dependency to `doctrine/persistence` 1
- FosUser should allow `doctrine/persistence` 2, it should be possible with the compatibility layer.

See https://github.com/FriendsOfSymfony/FOSUserBundle/issues/2996

Here is an example of a PR that could be done: https://github.com/doctrine/phpcr-odm/pull/812

Reply:
Yes, I agree. But FosUser is not maintain anymore :/

So until we're dropping this dependency, this is maybe the best solution we have

Reply:
Oh didn't know that. Then yes, that's probably the best solution.